### ISSUE TITLE
Generic typeDefs return incorrect methods

### ISSUE THREAD

**Issue by halildurmus** [halildurmus/winmd#60]
```dart
import 'package:winmd/winmd.dart';

void main() {
  final xmlTypeDef =
      MetadataStore.getMetadataForType('Windows.Data.Xml.Dom.IXmlNodeList')!;

  // IVectorView<IXmlNode>
  print(xmlTypeDef.interfaces.first.methods); // prints [Parse, TryParse]

  // IIterable<IXmlNode>
  print(xmlTypeDef.interfaces.last.methods); // prints [GetJsonStatus]

  // IVectorView<T>
  final vectorViewTypeDef = MetadataStore.getMetadataForType(
      'Windows.Foundation.Collections.IVectorView`1')!;

  print(vectorViewTypeDef.methods); // [GetAt, get_Size, IndexOf, GetMany]

  // IIterable<T>
  final iterableTypeDef = MetadataStore.getMetadataForType(
      'Windows.Foundation.Collections.IIterable`1')!;

  print(iterableTypeDef.methods); // [First]
}
```

In the above example, I'm printing out the methods from interfaces that `IXmlNodeList` implements. 

It prints [Parse, TryParse] for `IVectorView<IXmlNode>` and [GetJsonStatus] for `IIterable<IXmlNode>` which is incorrect. Instead, it should print `[GetAt, get_Size, IndexOf, GetMany]` for the former and `[First]` for the latter.

Note that this doesn't affect types like `...<T>` or `...<K, V>` (e.g. `IVectorView<T>`, `IMap<K, V>`). It only affects types whose generic arguments are specified (e.g. `IVectorView<IXmlNode>`).

**Comment by timsneath** [comment]
Note to self: `interface.first.token` is `0x1b000005`, which is a TypeSpec. This isn't being processed properly.

**Comment by timsneath** [comment]
Hmm... actually it does seem to be processed correctly (although the code was incorrectly allowing `IMetaDataImport.EnumMethods` and the like to be called with a token that is not a `TypeDef`. I've fixed that with #61.) 

Because these are typespecs, they don't fit neatly into a TypeDef. The following code should give the right results:

```dart
import 'package:winmd/winmd.dart';

void main() {
  final xmlTypeDef =
      MetadataStore.getMetadataForType('Windows.Data.Xml.Dom.IXmlNodeList')!;

  print(xmlTypeDef.interfaces.length);

  final typeSpecToken = xmlTypeDef.interfaces.first.token;
  print(typeSpecToken.toRadixString(16));

  // IVectorView<IXmlNode>
  print(xmlTypeDef.interfaces.first.methods); // prints [] now
  print(xmlTypeDef.interfaces.first.typeSpec?.type?.methods);
  // prints [GetAt, get_Size, IndexOf, GetMany]

  final foundType = TypeDef.fromTypeSpecToken(xmlTypeDef.scope, typeSpecToken);

  // Signature returned: [15, 12, 81, d1, 1, 12, 81, 61], which is correct
  // ELEMENT_TYPE_GENERICINST
  //   ELEMENT_TYPE_CLASS
  //     Windows.Foundation.Collections.IVectorView
  //     argcount 1
  //       ELEMENT_TYPE_CLASS
  //         Windows.Data.Xml.Dom.IXmlNode

  print(foundType.tokenType); // prints TokenType.typeSpec

  print(foundType.typeSpec?.name);
  // prints Windows.Foundation.Collections.IVectorView`1

  print(foundType.typeSpec?.typeArg?.name);
  // prints Windows.Data.Xml.Dom.IXmlNode

  // IIterable<IXmlNode>
  print(xmlTypeDef.interfaces.last.methods); // prints [] now
  print(xmlTypeDef.interfaces.last.typeSpec?.type?.methods); // prints [First]

  // IVectorView<T>
  final vectorViewTypeDef = MetadataStore.getMetadataForType(
      'Windows.Foundation.Collections.IVectorView`1')!;

  print(vectorViewTypeDef.methods); // [GetAt, get_Size, IndexOf, GetMany]

  // IIterable<T>
  final iterableTypeDef = MetadataStore.getMetadataForType(
      'Windows.Foundation.Collections.IIterable`1')!;

  print(iterableTypeDef.methods); // [First]
}
```

**Comment by halildurmus** [comment]
Thanks!

There is one more thing. `xmlTypeDef.interfaces.first.typeSpec?.type?.methods` returns generic methods:
```dart
import 'package:winmd/winmd.dart';

void main(List<String> args) {
  final xmlNodeList =
      MetadataStore.getMetadataForType('Windows.Data.Xml.Dom.IXmlNodeList')!;

  final vectorView = xmlNodeList.interfaces.first; // IVectorView<IXmlNode>
  final getAt = vectorView.typeSpec!.type!.methods.first; // GetAt

  // prints its BaseType as classVariableTypeModifier, instead it should be
  // classTypeModifier and its name should be Windows.Data.Xml.Dom.IXmlNode
  print(getAt.returnType.typeIdentifier);
}
```

It would be really great if there was a way to get a generic typeDef's methods with correct types based on the generic type's argument (IXmlNode). It would allow us to auto-generate `implementsMappers` for generic types such as `IVectorView<IXmlNode>` in `ixmlnodelist.dart` (currently, we ignore generic types while generating implementsMappers).

To do this I think we need to replace generic types in methods (these can be return types or parameters) with the parent type's argument (IXmlNode) as WinMD only exposes generic methods for these types. After that, these methods could be exposed through `xmlNodeList.interfaces.first.methods` (currently it returns `[]` with your latest patch).
